src/meta/src/rpc/server.rs

```diff
@@ -341,64 +345,166 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
 // Print the sleep intervals
 // use pseudo rand gen with seed
 mod tests {
+    use std::net::{IpAddr, Ipv4Addr};
+
+    use risingwave_pb::common::WorkerType;
+    use risingwave_rpc_client::MetaClient;
     use tokio::time::sleep;
 
     use super::*;
 
+    async fn setup_n_nodes(n: u16) -> Vec<(JoinHandle<()>, WatchSender<()>)> {
+        let meta_store = Arc::new(MemStore::default());
+
+        let mut node_controllers: Vec<(JoinHandle<()>, WatchSender<()>)> = vec![];
+        for i in 0..n {
+            // TODO: add pseudo random sleep here
+            let node = format!("node{}", i);
+            let err_msg = format!("Meta {} failed in setup", node);
+            let err_msg = err_msg.as_str();
+
+            let info = AddressInfo {
+                addr: node,
```

Comment:
`addr` is used for other components to connect to meta via this address. `listen_addr` is used for listen. We should config `addr` correctly and use it for election. Here in the test you can set them to the same.